next.config.cjs

```diff
@@ -0,0 +1,39 @@
+
+const nextConfig = {
+  reactStrictMode: true,
+  swcMinify: true,
+  images: { unoptimized: true },
+};
+
+const remarkMath = require('remark-math')
+const rehypeKatex = require('rehype-katex')
+
+const withMDX = require('@next/mdx')({
+  extension: /\.mdx?$/,
+  options: {
+    // If you use remark-gfm, you'll need to use next.config.mjs
+    // as the package is ESM only
+    // https://github.com/remarkjs/remark-gfm#install
+    remarkPlugins: [remarkMath],
+    rehypePlugins: [rehypeKatex],
+    // If you use `MDXProvider`, uncomment the following line.
+    // providerImportSource: "@mdx-js/react",
+  },
+})
+
+// module.exports = withMDX({
```

Comment:
Check in the docs if this one is really an alternative, i.e. `withMDX` "decorates" the existing configs, but retains those passed that it doesn't need to touch.

In case, I'd prefer to keep this version. If it's not working, iet's drop before merging.

Reply:
Yes it is. 
If I have understood, this is the common way to configure `mdx` into a `next` app. 
[Here](https://nextjs.org/docs/advanced-features/using-mdx#:~:text=Require%20the%20package%20and%20configure%20to%20support%20top%20level%20.mdx%20pages.%20The%20following%20adds%20the%20options%20object%20key%20allowing%20you%20to%20pass%20in%20any%20plugins%3A) the suggested setup procedure.

Reply:
Unfortunately, there isn't written what is the interplay of `withMDX()` (pay attention, it is the name you're assigning, no name comes from the package) and other options.

However, the package is extremely tiny (one single file of 40 lines), so I can confirm it from the source code:
- `withMDX` is already the result of calling a function, to which you passed the `plugin_options` https://github.com/vercel/next.js/blob/fbc98abab31a54dbc2b6c88a064b579a10f34871/packages/next-mdx/index.js#L2
- to `withMDX` you can pass any Next configuration https://github.com/vercel/next.js/blob/fbc98abab31a54dbc2b6c88a064b579a10f34871/packages/next-mdx/index.js#L3
- they will be copied into the result, but overwritten by generated settings https://github.com/vercel/next.js/blob/fbc98abab31a54dbc2b6c88a064b579a10f34871/packages/next-mdx/index.js#L22

[`Object.assign()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/assign) copies all the properties of a sequence of objects (second and further arguments) to the first one.
Later objects in the sequence may overwrite earlier ones.
So, since your `nextConfig` are the second argument, its elements might get overwritten by those of the generated third argument (but they won't since they are acting on different configs).

Reply:
I have fixed the `next.config.cjs` file, I think I can resolve this.